### PR TITLE
Do not overwrite Content-Type

### LINKED ISSUE #376: Content-Type set by other middleware gets overwritten
* Operating System: macOS 10.14.3
* Node Version: `v11.10.0`
* NPM Version: `6.8.0`
* webpack version: `4.29.6`
* webpack-dev-middleware Version: `3.6.0`

- [ ] This is a **feature** request
- [x] This is a **bug**

### Code

I am working with a custom middleware to serve static assets from their gzipped versions, like so:

```javascript
app.use(function (req, res, next) {
  if (req.url.endsWith('css')) {
    req.url = req.url + '.gz';
    res.set('Content-Encoding', 'gzip');
    res.set('Content-Type', 'text/css');
  }
  next()
});
```

Unfortunately, `webpack-dev-middleware` overwrites `Content-Type` regardless of whether or not it's been set already by another middleware. See

https://github.com/webpack/webpack-dev-middleware/blob/e5bd8f8d1b12543938601f1daab40e6173db7489/lib/middleware.js#L82

### Expected Behavior

In the resulting request: `Content-Type: text/css`

### Actual Behavior

In the resulting request: `Content-Type: application/gzip; charset=UTF-8`

### PR DESCRIPTION
**Pull request by dmohns**
**What kind of change does this PR introduce?**

When the Content-Type header is already present (for example it has been 
set by other middleware) we do not overwrite it.

**Did you add tests for your changes?**

Yes, added unit test to the suite.

**Summary**

Closes #376

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

File: lib/middleware.js [hunk 1]
```diff
@@ -79,7 +79,9 @@ module.exports = function wrapper(context) {
           contentType += '; charset=UTF-8';
         }
 
-        res.setHeader('Content-Type', contentType);
+        if (!res.getHeader('Content-Type')) {
+          res.setHeader('Content-Type', contentType);
+        }
         res.setHeader('Content-Length', content.length);
```
Comment (evilebottnawi) [inline lib/middleware.js]: What do you think should we add check here too? I think in theory it is also possible to set own `Content-Length`, don't known real use case, but still think what we should add check here too

Comment (dmohns) [reply lib/middleware.js]: This was my initial thought too. But I think matters are more complicated... It looks like ExpressJS itself sets the `Content-Length` 

https://github.com/expressjs/express/blob/3d10279826f59bf68e28995ce423f7bc4d2f11cf/lib/response.js#L194

In fact line 85 seems to be not functional at all.

Comment (evilebottnawi) [reply lib/middleware.js]: @dmohns good catch, can you create issue? Will be great if you help to use investigate this, let's merge this as is

Comment (dmohns) [reply lib/middleware.js]: 👍 Will do.

**Comment by dmohns** [comment]
Unfortunately, I just realized the test in this PR is "wrong". The headers get set *after* Content-Type already got overwritten. Hence we are not really testing the scope of this PR. I'll shoot a separate PR with a fix.

**Comment by evilebottnawi** [comment]
@dmohns :+1:

File: lib/middleware.js (hunk 1 above)
Comment (HeGangSZ) [reply lib/middleware.js]: hello，this commit make error，"TypeError: res.getHeader is not a function"，my node version is v10.15.2，how can i solve this problem

Comment (HeGangSZ) [reply lib/middleware.js]: > 👍 Will do.

hello，this commit make error，"TypeError: res.getHeader is not a function"，my node version is v10.15.2，how can i solve this problem

**Pull request merged**